React/.eslintrc.js

```diff
@@ -0,0 +1,347 @@
+        'ok',
+        'olympic',
+        'onclick',
+        'onhashchange',
+        'onreadystatechange',
+        'orderby',
+        'pageview',
+        'pangaea',
+        'param',
+        'params',
+        'perc',
+        'perf',
+        'pdf',
+        'pivotgrid',
+        'png', // DataGrid CustomEditors for image format
+        'popup',
+        'popups',
+        'pos',
+        'postfix',
+        'prec',
+        'pregrouped',
+        'prepend',
+        'prev',
+        'rangebar',
+        'readdir', // node js
+        'realtor',
+        'rect',
+        'req',
+        'resellers',
+        'resize',
+        'resizing',
+        'roomid',
+        'rowspan',
+        'rtl',
+        'sankey',
+        'sclass', // CSSClass
+        'scrollable',
+        'scrollbar',
+        'setted',
+        'skype',
+        'smp',
+        'sonee',
+        'sortable',
+        'sparkline',
+        'splashscreen',
+        'splinearea',
+        'sqlite',
+        'sqrt',
+        'src',
+        'stacktrace',
+        'startswith',
+        'steparea',
+        'stepline',
+        'str',
+        'stringify',
+        'strikethrough',
+        'submenu',
+        'substr',
+        'substring',
+        'subvalue',
+        'subvalues',
+        'svg',
+        'swfobject',
+        'tbody',
+        'th',
+        'td',
+        'theatre',
+        'timestamp',
+        'tmp',
+        'tooltip',
+        'tooltips',
+        'transpile',
+        'transpiler',
+        'treeview',
+        'ui', // User Interface
+        'ul',
+        'ungrouped',
+        'unlink',
+        'unordered',
+        'unselect', // unselectAll()
+        'unshift',
+        'uploader',
+        'uri',
+        'urlencoded',
+        'usa',
+        'util',
+        'utils',
+        'validator',
+        'viewport',
+        'volumne', // stockMarket, rename to volume
+        'vue',
+        'winloss',
+        'writeable', // knockout
+        'xlsx', // ExcelJS
+        'xhr',
+        'Xmla',
+        'xmla',
+        'youtube',
+        'ytd',
+        'zipcode',
+        'Serializer', // XMLSerializer
+        // VectorMap Demos
+        'Kosovo',
+        'Macau',
+        'Niue',
+        'Palau',
+      ]);
+      return spellcheckRule;
+    })(),
+    'class-methods-use-this': 0, // TODO warn (was error)
+    'func-names': 0, // TODO warn (was warn) >500
+    'import/extensions': 0,
+    'no-use-before-define': 0,
+    'prefer-destructuring': 0,
+    'no-param-reassign': ['error', { 'props': false }],
+    'no-only-tests/no-only-tests': 'error',
+  },
+  extends: [
+    'eslint:recommended',
+    'devextreme/javascript',
+    'devextreme/spell-check',
+  ],
+  overrides: [{
+    'files': [
+      '*.ts', '*.tsx'
+    ],
+    extends: ['devextreme/typescript'],
+    parserOptions: {
+      project: './tsconfig.json',
+      'createDefaultProgram': true,
+      'ecmaVersion': 6,
+    },
+    rules: {
+      // TODO consider this rules
```

Comment:
Where did we get these rules from?

And maybe we should extend some standard eslint rules of framework?